strawberryfields/parameters.py

```diff
@@ -19,335 +19,334 @@
-    If a TensorFlow object is involved, the result will always be a TensorFlow object.
+
+def par_str(p):
+    """String representation of the Operation parameter.
 
     Args:
-      x (Number, array, Tensor, Variable, RegRef, RegRefTransform): parameter value
+        p (Any): Operation parameter
+
+    Returns:
+        str: string representation
     """
-    # turn off the NumPy ufunc dispatching mechanism which is incompatible with our approach (see https://docs.scipy.org/doc/numpy-1.14.0/neps/ufunc-overrides.html)
-    # NOTE: Another possible approach would be to use https://docs.scipy.org/doc/numpy-1.14.0/reference/generated/numpy.lib.mixins.NDArrayOperatorsMixin.html
-    __array_ufunc__ = None
-
-    def __init__(self, x):
-        if isinstance(x, Parameter):
-            raise TypeError('Tried initializing a Parameter using a Parameter.')
-
-        #: set[RegRef]: parameter value depends on these RegRefs (if any), it can only be evaluated after the corresponding subsystems have been measured
-        self.deps = set()
-
-        # wrap RegRefs in the identity RegRefTransform
-        if isinstance(x, RegRef):
-            x = RegRefTransform(x)
-        elif isinstance(x, (numbers.Number, np.ndarray, _tf_classes, RegRefTransform)):
-            pass
-        else:
-            raise TypeError('Unsupported base object type: ' +
-                            x.__class__.__name__)
-
-        # add extra dependencies due to RegRefs
-        if isinstance(x, RegRefTransform):
-            self.deps.update(x.regrefs)
-        self.x = x  #: parameter value, or reference
-
-    def __str__(self):
-        if isinstance(self.x, numbers.Number):
-            return '{:.4g}'.format(self.x)
-        return self.x.__str__()
-
-    def __format__(self, format_spec):
-        return self.x.__format__(format_spec)
-
-    def evaluate(self):
-        """Evaluate the numerical value of a RegRefTransform-based parameter.
+    if isinstance(p, np.ndarray):
+        np.set_printoptions(precision=4)
+        return str(p)
+    if par_is_symbolic(p):
+        return str(p)
+    return '{:.4g}'.format(p)  # scalar parameters
 
-        Returns:
-          Parameter: self, unless self.x is a RegRefTransform in which case it is evaluated and a new Parameter instance is constructed on the result and returned
-        """
-        if isinstance(self.x, RegRefTransform):
-            return Parameter(self.x.evaluate())
-        return self
 
-    def _unwrap_and_cast(self, other):
-        """Unwrap Parameters and cast TensorFlow-type parameters to other dtypes during arithmetic.
 
-        The main reason we need this is that TensorFlow does not automatically promote int to float or float to complex but requires an explicit cast.
+#class MeasuredParameter(sympy.AtomicExpr):  # something is messed up in Sympy caching, maybe, frontend tests fail depending on their execution order
```

Comment:
What is current status of this line?

Reply:
Should be removed, inheriting from sympy.Symbol seems to work fine.